docs/core/concepts/results.md

```diff
@@ -1,50 +1,41 @@
+In the unconfigured case, all `State` objects produced by a flow have a basic `Result` object with only `Result.value` set. These base objects do not know how to write themselves to persistent storage.
 
 ```python
 >>> type(state._result)
 prefect.engine.result.Result
 >>> type(state._result.value)  # this is the type of your Task's return value
 dict
->>> type(state._result.safe_value)
-prefect.engine.result.NoResultType
->>> type(state._result.result_handler)
-NoneType
 ```
 
+You can instead configure your task to use a subclass of `Result` that aligned with a persistent storage backend. This allows you to turn on persistent forms of caching and checkpointing. (Learn more about that in [Caching and Persisting Data](persistence.md). These subclasses of `Result` always have four important attributes / methods:
+
+- `location`: the location the data should be stored to
+- `read()`: how to read from the storage backend
+- `write()`: how to write to the storage backend
+- `exists()`: how to determine if the location exists for this storage backend
+
+There are many different `Result` classes aligning with different storage backends depending on your needs, such as `GCSResult` and `S3Result`. See the whole list in the [API docs for results](../../api/latest/engine/results.md)
+
 ::: tip NoResult vs. a None Result
 To distinguish between a Task that runs but does not return output from a Task that has yet to run, Prefect
-also provides a `NoResult` object representing the _absence_ of computation / data. This is in contrast to a `Result`
-whose value is `None`. The `to_result` / `store_safe_value` methods, along with the `value` and `safe_value` attributes of `NoResult` all return the same `NoResult` object.
+also provides a `NoResult` object representing the _absence_ of computation / data. This is in contrast to a `Result` whose value is `None`.
 :::
 
-All Results also come equipped with `to_result` / `store_safe_value` methods. For Result objects which have a result handler attached, these methods read results from their handlers, and write results with their handlers, respectively.
+### Interacting with task `Result`s
 
-### Interacting with `Result`s
-
-The most common scenario in which a user might need to directly interact with a `Result` object in memory is when running flows locally. All Prefect `States` have `Result` objects built into them, which can be accessed via the private `_result` attribute. For convenience, the public `.result` property retrieves the underlying `Result`'s value.
+The most common scenario in which a user might need to directly interact with a `Result` object produced by the Prefect pipeline in memory is when running flows locally. All Prefect `States` have `Result` objects built into them, which can be accessed via the private `_result` attribute. For convenience, the public `.result` property retrieves the underlying `Result`'s value.
```

Comment:
The code block just below this has a [known bug](https://github.com/PrefectHQ/prefect/issues/2392) - let's take this opportunity to fix that code snippet

Reply:
Fixed in 8cc892379